godot-core/src/task/futures.rs

```diff
@@ -62,42 +70,49 @@ impl<T> Default for SignalFutureData<T> {
 }
 
 // Only public for itest.
-#[cfg_attr(feature = "trace", derive(Default))]
-pub struct SignalFutureResolver<R> {
-    data: Arc<Mutex<SignalFutureData<R>>>,
+pub struct SignalFutureResolver<R: IntoMaybeSend> {
+    data: Arc<Mutex<SignalFutureData<R::Target>>>,
 }
 
-impl<R> Clone for SignalFutureResolver<R> {
+impl<R: IntoMaybeSend> Clone for SignalFutureResolver<R> {
     fn clone(&self) -> Self {
         Self {
             data: self.data.clone(),
         }
     }
 }
 
-impl<R> SignalFutureResolver<R> {
-    fn new(data: Arc<Mutex<SignalFutureData<R>>>) -> Self {
+/// For itest to construct and teest a resolver.
+#[cfg(feature = "trace")]
+pub fn create_test_signal_future_resolver<R: IntoMaybeSend + Default>() -> SignalFutureResolver<R> {
+    SignalFutureResolver {
+        data: Default::default(),
```

Comment:
What is the actual value here? It's obscured by the generic `Default::default`...
Is it
```rs
Arc::new(Mutex::new(SignalFutureData::new(R::default()))
```
?

I wonder if it's better to have an unwieldy expression, but then explicitly see the value :thinking: